code/modules/admin/ipintel.dm

```diff
@@ -0,0 +1,224 @@
+/datum/ipintel
+	var/ip
+	var/intel = 0
+	var/cache = FALSE
+	var/cacheminutesago = 0
+	var/cachedate = ""
+	var/cacherealtime = 0
+
+/datum/ipintel/New()
+	cachedate = SQLtime()
+	cacherealtime = world.realtime
+
+/datum/ipintel/proc/is_valid()
+	. = FALSE
+	if(intel < 0)
+		return
+	if(intel <= config.ipintel_rating_bad)
+		if(world.realtime < cacherealtime + (config.ipintel_save_good * 60 * 60 * 10))
+			return TRUE
+	else
+		if(world.realtime < cacherealtime + (config.ipintel_save_bad * 60 * 60 * 10))
+			return TRUE
+
+/proc/get_ip_intel(ip, bypasscache = FALSE, updatecache = TRUE)
```

Comment:
the argument `ip` is passed unsanitized into an SQL query. Right now it's used only in one place, where the argument is provided by an internal BYOND proc. `sanitizeSQL` would be a good idea.

Reply:
Changed